### ISSUE TITLE
Unhandled Exception Error: Incorrect Side-by-Side Configuration

### ISSUE THREAD

**Issue by hairycactus** [SerGreen/Appacker#13]
**Possible Cause:** Missing/Incorrect Microsoft Visual C++ Redistributable Runtime Version

**Affected:** Appacker v1.3.8 (28 Dec 2020)

**OS:** Win 10 x64

When Appacker's `Pack` button is clicked, it shows a popup error notifying about **incorrect side-by-side configuration**.

If `Continue` is clicked, packing seems to be in progress (with the `Packing ... Please wait` button greyed out), but nothing is happening & Appacker uses 0% CPU. If `Quit` is clicked, Appacker exits immediately.

The same error occurs regardless of the source folder (of files) selected to be packed. Screenshot & details of the error message are at the bottom.

Windows Event Viewer shows that Appacker always fails at `verInfoLib.exe` (which Appacker self-extracts to the TEMP folder during every run) due to the missing dependency **Microsoft.VC80.MFC v8.0.50608.0 x86**.

The above appears to refer to the **Microsoft Visual C++ Redistributable 2005 SP1 x86 MFC Security Update** runtime library, specifically with the DLL version of v8.0.50608.0.

Microsoft only supplies the last-available [**MS Visual C++ 2005 SP1 MFC v8.0.50727.6195** (14 May 2011)](https://www.microsoft.com/en-us/download/details.aspx?id=26347). Placing the extracted runtime DLLs (`msvcr80.dll, msvcp80.dll, msvcm80.dll, vcomp.dll`) within Appacker's folder results in the same side-by-side configuration error when `Pack` is clicked.

Is it possible to provide Appacker pre-compiled with the required dependency DLLs, so that the EXE binary is self-contained, & doesn't rely on the host PC having some old, unavailable runtime library ? (I don't have any IT training or programming knowledge, so I have no idea how to compile Appacker from the source code.)

Also, for user awareness, it would be good to state upfront in the program description/ README.md all the dependencies required by Appacker.

Thanks !

### Screenshot:
![Appacker Error](https://i.imgur.com/aLejcmy.png)

### Appacker Popup Error:
```
Unhandled exception has occurred in your application.
The application failed to start because its side-by-side configuration is incorrect.
************** Exception Text **************
System.ComponentModel.Win32Exception (0x80004005): The application has failed to start because its side-by-side configuration is incorrect. Please see the application event log or use the command-line sxstrace.exe tool for more detail
   at System.Diagnostics.Process.StartWithShellExecuteEx(ProcessStartInfo startInfo)
   at System.Diagnostics.Process.Start()
   at System.Diagnostics.Process.Start(ProcessStartInfo startInfo)
   at Appacker.MainForm.StartPacking(String sourceAppFolder, String mainExePath, String destinationPath, String customIconPath, String customFileDescription, String launchArguments, Boolean selfRepackable, Boolean openUnpackDir, UnpackDirectory unpackDirectory, Boolean noGUI)
   at Appacker.MainForm.btnPack_Click(Object sender, EventArgs e)
   at System.Windows.Forms.Control.OnClick(EventArgs e)
   at System.Windows.Forms.Button.OnClick(EventArgs e)
   at System.Windows.Forms.Button.OnMouseUp(MouseEventArgs mevent)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ButtonBase.WndProc(Message& m)
   at System.Windows.Forms.Button.WndProc(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
```

### Windows Events Viewer: SideBySide Error (Event 33)
```
Activation context generation failed for "C:\Users\user\AppData\Local\Temp\1374be56-c8b3-42ed-bf1e-44dc9be0a7ba\verInfoLib.exe". Dependent Assembly Microsoft.VC80.MFC,processorArchitecture="x86",publicKeyToken="1fc8b3b9a1e18e3b",type="win32",version="8.0.50608.0" could not be found. Please use sxstrace.exe for detailed diagnosis.
```

**Comment by SerGreen** [comment]
Yes indeed, it seems that it has dependency on VC 2005 runtime, i didn't realize it before. VerInfoLib.exe is an external tool that i used to update description of packaged exe files and that tool is written in C++.  

But it doesn't have to be specifically v8.0.50608.0 of Visual C++ Redistributable, the one you can download from Microsoft will work just fine. It seems that you just don't have it installed on your PC. I tried uninstalling it from my PC and got the same error you have here, then i reinstalled it using the link you attached and it worked fine again.  

I couldn't manage to embed the required libraries into the app, but instead i updated Appacker to check for available VC 2005 Runtime and if not found it will show a warning. You can ignore the warning and it should pack even without C++ libraries, the only thing that will not work is changing FileDescription field, which is sometimes used by Windows as a name for the application (without C++ libs it will be just "Appacker package").  
Here's this new version 1.3.9: https://github.com/SerGreen/Appacker/releases/tag/v1.3.9  

Thank you for this detailed report!

**Comment by SerGreen** [comment]
Issue solved with a workaround in [v1.3.9](https://github.com/SerGreen/Appacker/releases/tag/v1.3.9) and a proper fix in [v1.3.10](https://github.com/SerGreen/Appacker/releases/tag/v1.3.10).
